### ISSUE TITLE
Unable to lookup available vm scale set network interfaces

### ISSUE THREAD

**Issue by selvasingh** [Azure/azure-sdk-for-java#1325]
@Checketts reported this issue

**Comment by anuchandy** [comment]
With the recent compute rollout targeting 2016-06-01, VMSS NIC List API started working, so fluent API for the same is a work item for next milestone. As a temporary workaround inner client can be used as used to fetch the NICs.

```java
File credFile = new File(System.getenv("AZURE_AUTH_LOCATION"));

ApplicationTokenCredentials credentials = ApplicationTokenCredentials.fromFile(credFile);
RestClient restClient = credentials.getEnvironment().newRestClientBuilder()
    .withLogLevel(HttpLoggingInterceptor.Level.BODY)
    .withCredentials(credentials)
    .build();

NetworkManagementClientImpl networkManagementClient = new NetworkManagementClientImpl(restClient);
networkManagementClient.withSubscriptionId(credentials.defaultSubscriptionId());
NetworkInterfacesInner networkInterfaces = networkManagementClient.networkInterfaces();
PagedList<NetworkInterfaceInner> vmssNics =
    networkInterfaces.listVirtualMachineScaleSetNetworkInterfaces("<resource-group-name>", "<scale-set-name>");
for (NetworkInterfaceInner nic : vmssNics) {
  System.out.println(nic.id());
}
```

**Comment by anuchandy** [comment]
Fixed in the latest package "1.0.0-beta4.1"
